setup.php

```diff
@@ -90,8 +90,17 @@ function plugin_init_flyvemdm() {
       ]);
    }
 
-   if ($plugin->isActivated('flyvemdm')) {
+   if(!$plugin->getFromDBbyDir('flyvemdm')){
+      // nothing more to do at this moment
+      return;
```

Comment:
I think you don't need to check if the method succeeded. GLPI automatically adds an entry in the DB if the plugin exists in the filesystem. The only case where a return will occur is when GLPI fails to write the row (mysql database with full filesystem or other severe failure not related to GLPI or its plugins)

Reply:
There are two checks, the first one is for not execute the rest of the checks because if that query return false and I don't check it, the autoload load will be include by the reason of the field state will return N/A which will be different of NOTACTIVATED